sdk/inc/azure/core/internal/az_hfsm_internal.h

```diff
@@ -0,0 +1,181 @@
+    az_event_policy_handler root_state,
+    _az_hfsm_get_parent get_parent_func,
+    az_event_policy* outbound_policy,
+    az_event_policy* inbound_policy);
+
+/**
+ * @brief Transition to a peer state.
+ *
+ * @note  Calling this as a result of AZ_HFSM_RETURN_HANDLE_BY_SUPERSTATE is supported.
+ *        The application developer is responsible with ensuring the hierarchy is properly defined
+ *        and no invalid calls are made.
+ *
+ * @param[in] h The #_az_hfsm to use for this call.
+ * @param[in] source_state The source state.
+ * @param[in] destination_state The destination state.
+ *
+ * @return An #az_result value indicating the result of the operation.
+ * @retval #AZ_OK Success.
+ */
+AZ_NODISCARD az_result _az_hfsm_transition_peer(
+    _az_hfsm* h,
+    az_event_policy_handler source_state,
+    az_event_policy_handler destination_state);
+
+/**
+ * @brief Transition to a sub state.
+ *
+ * @note  Calling this as a result of AZ_HFSM_RETURN_HANDLE_BY_SUPERSTATE is supported.
+ *        The application developer is responsible with ensuring the hierarchy is properly defined
+ *        and no invalid calls are made.
+ *
+ * @param[in] h The #_az_hfsm to use for this call.
+ * @param[in] source_state The source state.
+ * @param[in] destination_state The destination state.
+ *
+ * @return An #az_result value indicating the result of the operation.
+ * @retval #AZ_OK Success.
+ */
+AZ_NODISCARD az_result _az_hfsm_transition_substate(
+    _az_hfsm* h,
+    az_event_policy_handler source_state,
+    az_event_policy_handler destination_state);
+
+/**
+ * @brief Transition to a super state.
+ *
+ * @note  Calling this as a result of AZ_HFSM_RETURN_HANDLE_BY_SUPERSTATE is supported.
+ *        The application developer is responsible with ensuring the hierarchy is properly defined
+ *        and no invalid calls are made.
+ *
+ * @param[in] h The #_az_hfsm to use for this call.
+ * @param[in] source_state The source state.
+ * @param[in] destination_state The destination state.
+ *
+ * @return An #az_result value indicating the result of the operation.
+ * @retval #AZ_OK Success.
+ */
+AZ_NODISCARD az_result _az_hfsm_transition_superstate(
+    _az_hfsm* h,
+    az_event_policy_handler source_state,
+    az_event_policy_handler destination_state);
+
+/**
+ * @brief Synchronously sends an event to a HFSM object.
+ *
+ * @note The lifetime of the event's `data` argument must be guaranteed until this function returns.
+ *       All state handlers related to this event will execute on the current stack. In most cases
+ *       it is recommended that a queue together with a message pump is used as an intermediary to
+ *       avoid recursive calls.
+ *
+ * @param[in] h The #_az_hfsm to use for this call.
+ * @param[in] event The event being sent.
+ *
+ * @return An #az_result value indicating the result of the operation.
+ * @retval #AZ_OK Success
+ */
+AZ_NODISCARD az_result _az_hfsm_send_event(_az_hfsm* h, az_event event);
```

Comment:
Note: _az_hfsm const*

Reply:
Adding const here would lead to need to add it to `az_event_policy_handler`